### DISCUSSION TITLE
New Install, Cannot save page

### DISCUSSION THREAD

**Discussion by kobaz** [requarks/wiki#5653] [Error / Bug Report]
Steps:
New Page:
Path: /test
Visual Editor
Set Title, Desc: -> OK

Error when running job render-page: internal/fs/utils.js:269 throw err; ^ Error: EIO: i/o error, write at writeSync (fs.js:679:3) at SyncWriteStream._write (internal/fs/sync_write_stream.js:24:3) at doWrite (_stream_writable.js:403:12) at writeOrBuffer (_stream_writable.js:387:5) at SyncWriteStream.Writable.write (_stream_writable.js:318:11) at Object.init (/opt/wiki/server/core/config.js:29:20) at Object.<anonymous> (/opt/wiki/server/core/worker.js:12:16) at Module._compile (internal/modules/cjs/loader.js:999:30) at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10) at Module.load (internal/modules/cjs/loader.js:863:32) { errno: -5, syscall: 'write', code: 'EIO' }

![2022-09-16_10-30_1](https://user-images.githubusercontent.com/1695852/190663494-563f3d86-0951-4d50-84cb-269a3fb2cd89.png)
![2022-09-16_10-30](https://user-images.githubusercontent.com/1695852/190663497-1d6b19c0-b36d-447d-ba6f-6eb50623dab5.png)

**Comment by NGPixel** [comment]
Make sure the user running wiki.js has write access to its data folder.

**Reply by kobaz** [reply]
Probably not the best setup, but node server is running as root so that shouldn't be a problem.

**Reply by kobaz** [reply]
I also have another problem.

The write fails... and then if I try to save it again... I get this:

![2022-10-21_17-19](https://user-images.githubusercontent.com/1695852/197290464-4af68b77-0ebb-4651-9604-6f9796ccb918.png)

But here's the issue.  The page never got created and it's forever stuck being unable to use the same name.  It does not show up in the page tree and it cannot be removed.

**Reply by kobaz** [reply]
Something is very very wrong.  I even tried reinstalling.

When trying to make a new page I also get this randomly:
![2022-10-21_17-23](https://user-images.githubusercontent.com/1695852/197290975-ee024ab8-8fc5-41a5-a61b-11f74473e599.png)

**Reply by kobaz** [reply]
Other issues.

![2022-10-21_17-28](https://user-images.githubusercontent.com/1695852/197291590-3b407616-a7ac-45e0-b1c3-f2e2db2cbdd8.png)



Should I create new tickets for each of these?

**Reply by kobaz** [reply]
I upgraded to the latest wikijs (2.5.289)
I did not change any server-side permissions... I can now create pages.  Was this issue maybe fixed in the new version?


Do you offer paid support?  There's many issues I still need to overcome.

![2022-10-21_17-42](https://user-images.githubusercontent.com/1695852/197293265-2c7bec2e-f542-4b9e-95b8-999084e4dd81.png)

**Comment by JanZeman** [comment]
Hi @kobaz, I am afraid I went though the same time consuming process as you did. I saw all the error messages you posted here in a similar order😥 Please read here: https://github.com/requarks/wiki/discussions/5697.

Today I updated to 2.5.289 and the problem also disappeared. Then the problem with 'duplicate entry deleted' as in your case. That one seems to be solved by Administration > Utilities > Content > Rerender All Pages. So please give it a try and hopefully your last problem is solved as well...

**Comment by kobaz** [comment]
This fixed the problem!

**Reply by JanZeman** [reply]
I knew it :) No, I didn't but guessed it might be the same issue. I wish you have a long time in front of you without any troubles. I hope for the same as recently I gave more time to wiki maintenance than it saved me. I am sure it will start paying off again!

**Comment by kobaz** [comment]
So... final closing thoughts... I would think this should be left open as a bug report.  Here's what I would suggest:

When performing a fresh install or an upgrade -- it's common to get a lot of errors when making pages.  The error being 'Duplicate entry detected'.  Despite *not* making any kind of duplicate pages.. this error presents itself on a regular basis.

The fix for this is:
Administration > Utilities > Content > Rerender All Pages.

Can this be made to happen automatically?  Or even better, fix the root cause bug that causes this issue for new installations.
